### PR TITLE
feat: Add `--max-pods` support to the kubelet configuration

### LINKED ISSUE #2180: Limit nodes by amount of pods
Not too long ago I asked to add an option to limit nodes by pods limitation. 
I would like to know from which version this feature should be available. Thanks!

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes #2180 

**Description**

Allow user-defined `maxPods` value passed through the `kubeletConfiguration`

This change allows the following use cases:
1. Small instance types can have a higher pod density than given through the instance type pod density set by the cloud provider
2. Larger instance types can specify a smaller pod density than given through the instance type pod density set by the cloud provider 

**How was this change tested?**

* `make test`
* Manual validation of `--max-pods` passed through 

**Does this change impact docs?**
- [x] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note
None
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/tasks/pod-density.md [hunk 1]
```diff
@@ -22,7 +22,18 @@ Do not use the `max-pods` argument to kubelet. Karpenter is not aware of this va
...
+
+### Provisioner-Specific Pod Density
+
+Pod density can be configured at the provisioner level by specifying `maxPods` within the `.spec.kubeletConfiguration`. All nodes spawned by this provisioner will set this `maxPods` value on their kubelet and will account for this value during scheduling.
+
+See [Provisioner API Kubelet Configuration](../../provisioner/#maximum-pods) for more details.
+
+### Controller-Wide Pod Density
+
+Set the environment variable `AWS_ENI_LIMITED_POD_DENSITY: "false"` (or the argument  `--aws-eni-limited-pod-density=false`) in the Karpenter controller to allow nodes to host up to 110 pods by default.
```
Comment (bwagner5) [inline website/content/en/preview/tasks/pod-density.md:36]: Can you mark this flag as deprecated in favor of max-pods in the KubeletConfiguration?

File: website/content/en/preview/tasks/configuration.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ There are two main configuration mechanisms that can be used to configure Karpen
 |--|--|--|
 | AWS_DEFAULT_INSTANCE_PROFILE | \-\-aws-default-instance-profile | The default instance profile to use when provisioning nodes in AWS|
 | AWS_ENABLE_POD_ENI | \-\-aws-enable-pod-eni | If true then instances that support pod ENI will report a vpc.amazonaws.com/pod-eni resource (default = false)|
-| AWS_ENI_LIMITED_POD_DENSITY | \-\-aws-eni-limited-pod-density | Indicates whether new nodes should use ENI-based pod density (default = true)|
+| AWS_ENI_LIMITED_POD_DENSITY | \-\-aws-eni-limited-pod-density | Indicates whether new nodes should use ENI-based pod density (default = true) DEPRECATED: Use `.spec.kubeletConfiguration.maxPods` to set pod density on a per-provisioner basis|
```
Comment (bwagner5) [inline website/content/en/preview/tasks/configuration.md:19]: The table in this doc is generated with the `make docgen` target. You'll need to set this in `pkg/utils/options/options.go`.

Comment (jonathan-innis) [reply website/content/en/preview/tasks/configuration.md]: Updated with the correct change

**Review by bwagner5 — APPROVED** [review]
lgtm!

**Pull request merged**